In [4]:
#PROBLEM STATEMENT : PREDICT THE SURVIVAL OF TITANIC BASED ON GIVEN DATASET

In [5]:
import pandas as pd
data=pd.read_csv("titanic.csv")

In [6]:
len(data)

891

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 66.2+ KB


In [10]:
data.shape

(891, 12)

Data Dictionary


Survived: 0 = No, 1 = Yes 
pclass: Ticket class 1 = 1st, 2 = 2nd, 3 = 3rd 
sibsp: # of siblings / spouses aboard the Titanic 
parch: # of parents / children aboard the Titanic 
ticket: Ticket number 
cabin: Cabin number 
embarked: Port of Embarkation C = Cherbourg, Q = Queenstown, S = Southampton 

Total rows and columns
We can see that there are 891 rows and 12 columns in our training dataset.

we can see that AGE is missing in many rows ie,out of 891 rows only 714 rows have the age as not null
Similarly, Cabin values are also missing in many rows. Only 204 out of 891 rows have Cabin values.

In [11]:
data.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

There are 177 rows with missing Age, 687 rows with missing Cabin and 2 rows with missing Embarked information.

VISUALIZATION

In [12]:
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set() # setting seaborn default for plots

Bar Chart for  the Categories


Pclass

Sex

SibSp ( # of siblings and spouse)

Parch ( # of parents and children)

Embarked

Cabin